### PR TITLE
remove old types from transport and use packp

### PR DESCRIPTION
**Pull request by smola**
* move Upload UploadPackRequest to protocol.
* rename UlReq to UploadRequest
* move AdvRefs Encoder/Decoder to Encode/Decode methods.
* move UploadRequest Encoder/Decoder to Encode/Decode methods.
* remove transport.UploadPackInfo in favor of packp. AdvRefs.
* rename Remote.Ref/Refs to Reference/References.

### CODE REVIEW

File: plumbing/protocol/packp/advrefs.go [hunk 1]
```diff
@@ -38,3 +42,59 @@ func NewAdvRefs() *AdvRefs {
...
+	for _, symref := range ar.Capabilities.Get("symref").Values {
+		chunks := strings.Split(symref, ":")
+		if len(chunks) != 2 {
+			err := fmt.Errorf("bad number of `:` in symref value (%q)", symref)
+			return plumbing.NewUnexpectedError(err)
+		}
+		name := plumbing.ReferenceName(chunks[0])
+		target := plumbing.ReferenceName(chunks[1])
+		ref := plumbing.NewSymbolicReference(name, target)
+		_ = s.SetReference(ref)
```
Comment (mcuadros) [inline plumbing/protocol/packp/advrefs.go]: get the error and return it please

File: plumbing/protocol/packp/advrefs_decode.go [hunk 2]
```diff
@@ -45,10 +46,10 @@ func (d *AdvRefsDecoder) Decode(v *AdvRefs) error {
 	return d.err
 }
 
-type decoderStateFn func(*AdvRefsDecoder) decoderStateFn
+type decoderStateFn func(*advRefsDecoder) decoderStateFn
 
 // fills out the parser stiky error
-func (d *AdvRefsDecoder) error(format string, a ...interface{}) {
+func (d *advRefsDecoder) error(format string, a ...interface{}) {
 	d.err = fmt.Errorf("pkt-line %d: %s", d.nLine,
```
Comment (mcuadros) [inline plumbing/protocol/packp/advrefs_decode.go]: Errorf, Sprintf? can we make only one call?

File: plumbing/protocol/packp/advrefs_encoder.go [hunk 3]
```diff
@@ -9,26 +9,29 @@ import (
 	"gopkg.in/src-d/go-git.v4/plumbing/format/pktline"
 )
 
-// An AdvRefsEncoder writes AdvRefs values to an output stream.
-type AdvRefsEncoder struct {
+// Encode writes the AdvRefs encoding of v to the stream.
```
Comment (mcuadros) [inline plumbing/protocol/packp/advrefs_encoder.go]: v?

File: plumbing/protocol/packp/advrefs.go (hunk 1 above)
Comment (smola) [inline plumbing/protocol/packp/advrefs.go]: Done.

**Pull request merged**